test/test_rack_handler.rb

```diff
@@ -360,5 +365,26 @@ def test_bin
         end
       end
     end
+
+    def test_rackup1
+      pid = nil
+      # JRuby & TruffleRuby take a long time using IO.popen
+      skip_unless :mri
+      #io = IO.popen "RUBY_OPT='-rrack/handler' rackup -p 0"
+      io = IO.popen "RUBYOPT='-rbundler/setup -rrack/version -rrack/handler -rrackup -rrack/handler/puma' ruby -e 'puts Rackup::VERSION'"
+      io.wait_readable 2
+      sleep 0.7
+      pid = io.pid
+      log = io.sysread 2_048
+      assert_match(/\A1\./, log)
+    ensure
+      if pid
+        if Puma::IS_WINDOWS
+          `taskkill /F /PID #{pid}`
```

Comment:
Can you not use `Process.kill`?

Reply:
I haven't checked for a while, but Windows doesn't handle signals the same as *nix or macos.

I just copied the basic code from `test_bin`, which I think I wrote.  I'll have a look.  Might be tomorrow AM (US).

Reply:
@ioquatix

Thanks.  I looked at this, and with possible errors being raised, or extra messages to the CI log, this is probably the easiest way to shut a process down, especially with multiple OS's.

But while looking at it, I noticed that I had a bash command line for `IO.popen`, so I updated it to work with Windows by using the `env` parameter for passing `RUBYOPT`.

We don't test Rack2/Rackup on Windows in CI.